### ISSUE TITLE
HttpRequestMessage in .NET Framework uses _headers not headers

### ISSUE THREAD

**Issue by ghost1face** [TalAloni/StandardSocketsHttpHandler#6]
When using .NET Framework 4.6.1 and `System.Net.Http 4.3.4` seems to use `_headers` instead of `headers` which causes a `NullReferenceException` since the field cannot be found:

https://github.com/TalAloni/StandardSocketsHttpHandler/blob/ddb13933d61cb2580c435bb98ec77467a2ae9b2e/StandardSocketsHttpHandler/Net/Http/Extensions/HttpRequestMessageExtensions.cs#L12

Here's my stack trace:
```   
at System.Net.Http.HttpRequestMessageExtensions.HasHeaders(HttpRequestMessage request)
   at System.Net.Http.StandardSocketsHttpHandler.ValidateAndNormalizeRequest(HttpRequestMessage request)
   at System.Net.Http.StandardSocketsHttpHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.DelegatingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.<SendAsync>d__2.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendAsync>d__2.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

ILSpy of `System.Net.Http 4.3.4`:
![image](https://user-images.githubusercontent.com/2773690/125332099-ce9b4500-e316-11eb-98c0-9bd7082822d8.png)

I'm not familiar with the exact reason you put this in place, but should we check for both fields until we find one instead for backward compatibility?

**Comment by TalAloni** [comment]
.NET Framework 4.6.1 is not supported.
I'm assuming this is not the only issue with .NET Framework 4.6.1.

**Comment by TalAloni** [comment]
I've added a fallback to avoid this exception in .NET Framework 4.6.1, available in v2.2.0.3
